libethereum/BlockChain.cpp

```diff
@@ -856,6 +856,8 @@ void BlockChain::rescue(OverlayDB& _db)
 				break;
 		}
 		catch (...) {}
+		if (l == 0)
```

Comment:
I have a test where this cycle in this function goes l = 0;  l--;   l = 497....  
this way it would run endless.